### PR TITLE
feat: validate path parameters

### LINKED ISSUE #15: Schema "path" parameters are not validated
Given the following schema:

```yml
paths:
  /users/{id}:
    get:
      summary: view a user
      operationId: user.view
      produces:
      - application/json
      parameters:
      - in: path
        name: id
        description: user id
        required: true
        type: integer
        minimum: 1
```

The `{id}` parameter should be validated against the request path.

### PR DESCRIPTION
**Pull request by shadowhand**
fixes #15

### CODE REVIEW

**Comment by guillemcanal** [comment]
We are up to a good start but still, it won't work as is since path parameters need to be coerced before being validated.

Values extracted from the URI template are strings, if you require integer, the validation will fail.

I'll upgrade the `justinrainbow/jsonschema`dependency to a more up to date version which take care of type coercion.

**Comment by shadowhand** [comment]
@guillemcanal 🤦‍♂️ I didn't even think of that. Thanks!

**Comment by shadowhand** [comment]
@guillemcanal this should be cover all concerns now, as it tests both failure and success outcomes.

File: src/Factory/SwaggerSchemaFactory.php [hunk 1]
```diff
@@ -73,12 +73,31 @@ protected function resolveSchemaFile($schemaFile)
...
             new UriResolver()
         );
 
-        return $refResolver->resolve($schemaFile);
+        $schema = $schemaStorage->getSchema($schemaFile);
+
+        // JsonSchema normally defers resolution of $ref values until validation.
+        // That does not work for us, because we need to have the complete schema
+        // to build definitions.
+        $this->expandSchemaReferences($schema, $schemaStorage);
```
Comment (shadowhand) [inline src/Factory/SwaggerSchemaFactory.php]: This was annoying to track down but my YAML schema now works properly.

Comment (guillemcanal) [reply src/Factory/SwaggerSchemaFactory.php]: Nicely stopped. The new public API of JSON Schema is confusing as Hell. I've experimented quite a bit the the library and I had some ideas to improve the implementation. But this solution will do nicely.

**Pull request merged**